src/components/LetterSelection/index.jsx

```diff
@@ -77,51 +97,96 @@ const reducer = (state, action) => {
 
   useEffect(() => {
     dispatch({
       type: types.initialize,
-      payload: word,
+      payload: { word: word, corrects: corrects },
     });
   }, [word]);
 
   return (
     <>
-      {letterState.clickedLetters.map((clickedLetter, index) => {
-        return <div key={index}>{clickedLetter}</div>;
-      })}
+      <div className={styles.clicked_letters_container}>
+        {letterState.clickedLetters.map((clickedLetter, index) => {
+          return (
+            <div
+              className={`${styles.clicked_letter} ${
+                letterState.correct ? styles.correct : ""
+              }`}
+              key={index}
+            >
+              {clickedLetter}
+            </div>
+          );
+        })}
+      </div>
 
-      {/* Buttons to select the letters */}
-      {letterState.selectableLetters.map((selectableLetter, index) => {
-        return (
+      <div className={styles.buttons_container}>
+        {/* Buttons to select the letters */}
+        <div className={styles.selectable_letters_container}>
+          {letterState.selectableLetters.map((selectableLetter, index) => {
+            return (
+              <button
+                key={index}
+                className={`${styles.selectable_letter} ${
+                  letterState.clickedLetterIndexes.includes(index) ||
+                  letterState.correct
+                    ? styles.disabled
+                    : ""
+                } ${
+                  letterState.clickedLetters.length ===
+                  letterState.clickedLetterIndexes.length
+                    ? styles.filled
+                    : ""
+                }`}
+                onClick={() =>
+                  dispatch({
+                    type: types.addLetter,
+                    payload: index,
+                  })
+                }
+              >
+                {selectableLetter}
+              </button>
+            );
+          })}
+        </div>
+        <div className={styles.utility_buttons_container}>
+          <button
+            className={`${styles.backspace} ${
+              letterState.correct ? styles.disabled : ""
+            }`}
+            onClick={() => dispatch({ type: types.removeLetter })}
+          >
+            <TiBackspace />
+          </button>
           <button
-            key={index}
+            className={`${styles.submit} ${
+              letterState.correct ? styles.disabled : ""
+            }`}
             onClick={() =>
-              dispatch({
-                type: types.addLetter,
-                payload: index,
+              dispatchGame({
+                type: typesGame.checkWord,
+                payload: {
+                  clickedLetters: letterState.clickedLetters,
+                  letterDispatch: dispatch,
+                  setCorrectInLetters: types.setCorrectInLetters,
+                },
               })
             }
           >
-            {selectableLetter}
+            <GrReturn />
```

Comment:
More of a style choice than a coding problem, but I think this should be above the backspace. I see the logic for putting it below, but I think it feels better above.

Reply:
What should be above the backspace? The submit button?

Reply:
Correct. Is this not the right spot that is. I just assumed it was the Green Return symbol.

Reply:
I disagree that the backspace button should be below the return symbol. (i.e. from physical keyboards to on-screen touch ones, backspace is always above the return key)

Reply:
We'll I guess duel to the death is the only way to resolve this empasse.